notifications_utils/decorators.py

```diff
@@ -14,3 +20,116 @@ def wrapper(*args, **kwargs):
         return wrapper
 
     return decorator_feature_flag
+
+
+# Helper function to chunk a list
+def chunk_iterable(iterable_collection: Iterable, chunk_size: int) -> Generator:
+    """Helper function to chunk an iterable collection in preparation for parallel processing.
+
+    Args:
+        iterable_collection (Iterable): The collection to be chunked
+        chunk_size (int): The size of each chunk
+
+    Yields:
+        list: The next chunk of the iterable
+    """
+    iterable = iter(iterable_collection)
+    while True:
+        chunk = list(islice(iterable, chunk_size))
+        if not chunk:
+            break
+        yield chunk
+
+
+def control_chunk_and_worker_size(data_size=None, chunk_size=None, max_workers=None):
```

Comment:
_The parallelization related functions could be in their own modules, i.e. `parallel` or `iterable`, and the `decorators` module can act as a front-end to the developer.. but the parallel utilities would be available for other usage. Just a thought on how to cut down the logic, not necessary if you don't feel it.

Reply:
I agree, this could be useful in other contexts, I'll pull it out into it's own module along with the `chunk_iterable` method.